### ISSUE TITLE
Errors about invalid references are not associated with a project

### ISSUE THREAD

**Issue by davkean** [dotnet/project-system#1160]
1. Create a new Console App targeting .NETCoreApp1.0
2. Create a new Console App targeting .NETCoreApp1.1
3. Reference 2 from 1

```
Severity	Code	Description	Project	File	Line	Suppression State
Error		Project ConsoleApp49 is not compatible with netcoreapp1.0 (.NETCoreApp,Version=v1.0). Project ConsoleApp49 supports: netcoreapp1.1 (.NETCoreApp,Version=v1.1)			0	
Error		One or more projects are incompatible with .NETCoreApp,Version=v1.0.			0	
```

![image](https://cloud.githubusercontent.com/assets/1103906/21922976/99dc6808-d9c5-11e6-92a1-bfe6254b3bfc.png)

Not sure if this is our bug or SDK, but these should be associated with a project.

**Comment by davkean** [comment]
Maybe this is NuGet outputting this? 

Can we stop NuGet from adding errors to the error list, and instead use our errors from Resolve the reference?

**Comment by srivatsn** [comment]
@natidea @emgarten where did we land on who's outputting the errors for unresolved package refs?

**Comment by natidea** [comment]
I think these errors are coming from NuGet. The plan is for NuGet to add these to the assets file, and we can raise to the error list as needed. 

Right now NuGet emits unresolved package refs to error list, and our task adds some errors to the solution explorer where applicable (but suppresses messages to the error list for now, tracked by https://github.com/dotnet/sdk/issues/585)

**Comment by davkean** [comment]
Where's the bug tracking this work?

**Comment by davkean** [comment]
The other thing I'd like to see is the errors around, references, package references and project references to be unified so that they look and feel similar. Right now it the error message for packages is really hard to parse:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error		Package System 4.1.311.2 is not compatible with netcoreapp1.0 (.NETCoreApp,Version=v1.0). Package System 4.1.311.2 supports: net451 (.NETFramework,Version=v4.5.1)			0	

```

**Comment by davkean** [comment]
I also don't understand this, Dependency node makes it out to seem this is a bug:

![image](https://cloud.githubusercontent.com/assets/1103906/22102089/3f15f60e-de8a-11e6-91ae-f2b2c1f95a5c.png)

Whereas, error list tells me the correct story:

```
Warning		Dependency specified was System (>= 1.0.0) but ended up with System 4.1.0311.2.			0	
```

**Comment by natidea** [comment]
The NuGet work is tracked at https://github.com/NuGet/Home/issues/1599

The dependency warnings were copied from the CLI (see https://github.com/dotnet/sdk/issues/26#issuecomment-262709524). They may have slightly different wordings from NuGet, or I may have selected the wrong one for this scenario.

**Comment by abpiskunov** [comment]
yes , message in the tree should be same as warning in error tree (it should come form the sdk, tree just display's it). Another thing is we need to add a correct icon for warnings instead of a question mark (that i will do)

**Comment by srivatsn** [comment]
Tracking this as part of https://github.com/dotnet/project-system/issues/1660
